Profpoint trap event collector

daryakoval opened the pull request:
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-3170

## Description
Event collector 

## Minimum version of Cortex XSOAR
- [x] 6.8.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

dantavori commented on Packs/ProofpointThreatResponse/Integrations/ProofpointThreatResponseEventCollector/README.md:
```diff
@@ -0,0 +1,51 @@
+Use the Proofpoint Threat Response integration to orchestrate and automate incident response.
+
+## Configure Proofpoint Threat Response Event Collector on Cortex XSOAR
```
```suggestion
## Configure Proofpoint Threat Response Event Collector on Cortex XSIAM
```

dantavori commented on Packs/ProofpointThreatResponse/ReleaseNotes/1_0_13.md:
```diff
@@ -0,0 +1,8 @@
+
+#### Integrations
+##### New: Proofpoint Threat Response Event Collector
+Use the Proofpoint Threat Response integration to orchestrate and automate incident response. (Available from Cortex XSOAR 6.8.0).
+
+#### Modeling Rules
+##### New: Proofpoint Threat Response Modeling Rule
+Added the *Email* modeling rule. (Available from Cortex XSOAR 6.8.0).
```
```suggestion
Use the Proofpoint Threat Response integration to orchestrate and automate incident response. (Available from Cortex XSIAM 3.3.0).

#### Modeling Rules
##### New: Proofpoint Threat Response Modeling Rule
Added the *Email* modeling rule. (Available from Cortex XSIAM 3.3.0).
```

dantavori commented on Packs/ProofpointThreatResponse/Integrations/ProofpointThreatResponseEventCollector/ProofpointThreatResponseEventCollector_description.md:
```diff
@@ -0,0 +1,10 @@
+
+  Settings -> API Key -> Add api key
+  
+  Note: This is a beta Integration, which lets you implement and test pre-release software. Since the integration is beta, it might contain bugs. Updates to the integration during the beta phase might include non-backward compatible features. We appreciate your feedback on the quality and usability of the integration to help us identify issues, fix them, and continually improve.
+  
```
```suggestion  
```

dantavori commented on Packs/ProofpointThreatResponse/Integrations/ProofpointThreatResponseEventCollector/ProofpointThreatResponseEventCollector_description.md:
```diff
@@ -0,0 +1,10 @@
+  
+  Note: This is a beta Integration, which lets you implement and test pre-release software. Since the integration is beta, it might contain bugs. Updates to the integration during the beta phase might include non-backward compatible features. We appreciate your feedback on the quality and usability of the integration to help us identify issues, fix them, and continually improve.
+  
+- The ***proofpoint-tr-ingest-alert*** command requires some JSON format arguments. Please make sure the inputed JSONs have uniqe characters (such as `"`) escaped before entering them (e.g. `\"`).
+- For more information regarding the JSON objects, please see Proofpoint TRAP documentation under "JSON Alert Source 2.0".
```
```suggestion
```

dantavori commented on Packs/ProofpointThreatResponse/Integrations/ProofpointThreatResponseEventCollector/ProofpointThreatResponseEventCollector.py:
```diff
@@ -0,0 +1,382 @@
+        'last_fetched_incident_id': last_fetched_id
+    }
+
+    demisto.info(f'extracted {len(incidents)} events')
```
```suggestion
    demisto.debug(f'Fetched {len(incidents)} events')
```

dantavori commented on Packs/ProofpointThreatResponse/Integrations/ProofpointThreatResponseEventCollector/ProofpointThreatResponseEventCollector.py:
```diff
@@ -0,0 +1,382 @@
+        new_incidents = get_new_incidents(client, request_params, last_fetched_id)
+        incidents_list.extend(new_incidents)
+
+        demisto.info(
```
```suggestion
        demisto.debug(
```

dantavori commented on Packs/ProofpointThreatResponse/Integrations/ProofpointThreatResponseEventCollector/ProofpointThreatResponseEventCollector.py:
```diff
@@ -0,0 +1,382 @@
+        if not last_fetched_id.get(state):
+            last_fetched_id[state] = '0'
+
+    incidents = []
```
```suggestion
    events = []
```

dantavori commented on Packs/ProofpointThreatResponse/Integrations/ProofpointThreatResponseEventCollector/ProofpointThreatResponseEventCollector.py:
```diff
@@ -0,0 +1,382 @@
+
+class Client(BaseClient):
+    """
+    Client will implement the service API, and should not contain any Demisto logic.
```
```suggestion
    Client will implement the service API, and should not contain any logic.
```

dantavori commented on Packs/ProofpointThreatResponse/Integrations/ProofpointThreatResponseEventCollector/ProofpointThreatResponseEventCollector.yml:
```diff
@@ -0,0 +1,153 @@
+  type: 9
+  required: true
+  hiddenusername: true
+- display: Trust any certificate (not secure)
+  name: insecure
+  required: false
+  type: 8
+- defaultvalue: 'false'
+  display: Use system proxy settings
+  name: proxy
+  required: false
+  type: 8
```
please move to bottom of the params

dantavori commented on Packs/ProofpointThreatResponse/Integrations/ProofpointThreatResponseEventCollector/ProofpointThreatResponseEventCollector.yml:
```diff
@@ -0,0 +1,153 @@
+  name: post_url_id
+  required: false
+  type: 0
+description: Use the Proofpoint Threat Response integration to orchestrate and automate incident response.
```
please verify this description with a TPM

daryakoval replied:
Meital is out of the office.  This is the description of the integration that fetches incidents.

dantavori commented on Packs/ProofpointThreatResponse/Integrations/ProofpointThreatResponseEventCollector/ProofpointThreatResponseEventCollector_description.md:
```diff
@@ -0,0 +1,6 @@
+
+  Settings -> API Key -> Add api key
+  
+- If the **fetch-incidents** result in timeout, please consider changing **fetch-delta** and **fetch-limit** parameters 
+which will limit the api calls and the result.
```
```suggestion
**Note:** if the **fetch-events** command results in a timeout, consider changing **fetch_delta** and **fetch_limit** parameters to reduce the number of API calls and results.
```

dantavori commented on Packs/ProofpointThreatResponse/ModelingRules/ProofpointThreatResponseModelingRules/ProofpointThreatResponseModelingRules.yml:
```diff
@@ -0,0 +1,9 @@
+tags: proofpoint
+rules: ''
+schema: ''
+to_xdr_version: 3.4.9
+toversion: 6.9.9
```
```suggestion
```
as Shahaf said last week, if the issues with the new modeling rules are not resolved until you are ready to merge, let's remove them and we'll add them later

dantavori commented on Packs/ProofpointThreatResponse/pack_metadata.json:
```diff
@@ -1,8 +1,8 @@
     "description": "Use the Proofpoint Threat Response integration to orchestrate and automate incident response.",
     "support": "xsoar",
-    "currentVersion": "1.0.13",
+    "currentVersion": "1.0.14",
```
should be major update for new integration - 2.0.0

dantavori commented on Packs/ProofpointThreatResponse/Integrations/ProofpointThreatResponseEventCollector/ProofpointThreatResponseEventCollector.yml:
```diff
@@ -0,0 +1,153 @@
+category: Email Gateway
```
shouldn't it be `Analytics & SIEM`?

daryakoval replied:
No, it is the same as Integration.

dantavori commented on Packs/ProofpointThreatResponse/Integrations/ProofpointThreatResponseEventCollector/ProofpointThreatResponseEventCollector.py:
```diff
@@ -0,0 +1,402 @@
+    params = demisto.params()
+
+    api_key = params.get('credentials', {}).get('password')
+    base_url = demisto.params().get('url')
```
```suggestion
    base_url = params.get('url')
```

dantavori commented on Packs/ProofpointThreatResponse/Integrations/ProofpointThreatResponseEventCollector/ProofpointThreatResponseEventCollector.py:
```diff
@@ -0,0 +1,402 @@
+    return events, last_run
+
+
+def get_events_from_incidents(incidents):
```
please add a docstring

dantavori reviewed (commented):
looks great, please assign a doc reviewer
I will probably approve after the demo next week

ShirleyDenkberg commented:
@daryakoval Shouldn't there be a pack README?
@dantavori Doc review completed.

daryakoval commented:
Force merge needed: Proofpoint pack was only for "xsoar" marketplace. I added event collector and added to the pack metadata "marketplacev2" too. 
The build fails on the "pre-update" step, since before the update there is no such pack in the marketplacev2, and after the update the "post-update" passes.
